src/service/recruit/board.ts

```diff
@@ -0,0 +1,41 @@
+import { EndPoint, axiosInstance, axiosAuthInstance } from '..';
+import { ListResult, FilterData } from '../../types';
+
+interface FilterItem {
+	filterState: FilterData;
+	isLoggedIn: boolean;
+}
+
+export const getPostList = async ({ filterState, isLoggedIn }: FilterItem) => {
+	try {
+		const queryString = Object.entries(filterState)
+			.filter(([key, value]) => {
+				if (key === 'keyword') {
+					return value !== null && value !== undefined && value !== '';
+				} else if (Array.isArray(value)) {
+					return value.length > 0;
+				} else {
+					return value !== null && value !== undefined;
+				}
+			})
+			.map(([key, value]) => {
+				if (Array.isArray(value)) {
+					return value.map(v => `${key}=${encodeURIComponent(v)}`).join('&');
+				} else {
+					return `${key}=${encodeURIComponent(value)}`;
+				}
+			})
+			.join('&');
+
+		const url = `${EndPoint.RECRUITMENT_BOARD.list}${queryString ? `?${queryString}` : ''}`;
+		if (isLoggedIn) {
+			const response = await axiosAuthInstance.get<ListResult>(url);
+			return response;
+		} else {
+			const response = await axiosInstance.get<ListResult>(url);
+			return response;
+		}
+	} catch (error) {
+		console.error(error);
+	}
+};
```

Comment:
```suggestion
export const getPostList = async ({ filterState, isLoggedIn }: FilterItem) => {
	try {
		if (isLoggedIn) {
			const response = await axiosAuthInstance.get<ListResult>(EndPoint.RECRUITMENT_BOARD.list, {
				params: {
					...filterState,
				},
			});
			return response;
		} else {
			const response = await axiosInstance.get<ListResult>(url);
			return response;
		}
	} catch (error) {
		console.error(error);
	}
};
```

axios에서 쿼리스트링을 이용하고 싶을 때, params로 설정하면 간편하게 할 수 있습니다!

Reply:
하... 미쳤네요ㅋㅋㅋㅋ 뻘짓..했네..ㅋㅋㅋㅋ큐ㅠㅠㅠ
너무 사긴데..?

Reply:
수연님! 근데 이게 안 되는게 있어서 일단은 복구해놓았습니다ㅠㅠ 
상세정보 필터링이 안 먹히네요ㅠ

Reply:
!!!! 상세 정보 필터링이 안먹히나요?? 헉.. 왜지??!!